app/src/main/java/com/mashup/ui/mypage/MyPageViewModel.kt

```diff
@@ -56,70 +57,69 @@ class MyPageViewModel @Inject constructor(
-                                    cumulativeScore = score.cumulativeScore,
-                                    score = score.score,
-                                    detail = score.scheduleName,
-                                    date = score.date
-                                )
-                            )
-                        }
+                    } else {
+                        attendanceEmpty(profile)
                     }
-                } else {
-                    list.addAll(
-                        listOf(
-                            AttendanceModel(
-                                0, MyPageAdapterType.TITLE, profile, null, null
-                            ),
-                            AttendanceModel(
-                                0, MyPageAdapterType.SCORE, profile, null, null
-                            ),
-                            AttendanceModel(
-                                0, MyPageAdapterType.LIST_NONE, profile, null, null
-                            )
-                        )
-                    )
-
-                }
-                _attendanceList.postValue(list)
+                _attendanceList.postValue(attendanceItem)
             } else {
                 handleErrorCode(response.code)
             }
         } catch (ignore: Exception) {
         }
     }
 
+    private fun attendanceScoreList(
+        profile: Profile,
+        filterItem: List<ScoreHistoryResponse>
+    ): MutableList<AttendanceModel> {
+
+        val attendanceItem = mutableListOf<AttendanceModel>()
+        attendanceItem.addAll(myPageHeader(profile))
+
+        filterItem.forEach {
+            attendanceItem += AttendanceModel(
+                id = attendanceItem.size,
+                myPageType = MyPageAdapterType.LIST_LEVEL,
+                generationNum = it.generationNumber
+            )
+            attendanceItem += it.scoreDetails.map { score ->
+                AttendanceModel(
+                    id = attendanceItem.size,
+                    myPageType = MyPageAdapterType.LIST_ITEM,
+                    generationNum = it.generationNumber,
+                    activityHistory = ActivityHistory(
+                        scoreName = score.scoreName,
+                        attendanceType = AttendanceType.getAttendanceType(score.scoreType),
+                        cumulativeScore = score.cumulativeScore,
+                        score = score.score,
+                        detail = score.scheduleName,
+                        date = score.date
+                    )
+                )
+            }
+        }
+        return attendanceItem
+    }
+
+    private fun myPageHeader(profile: Profile) = listOf(
+        AttendanceModel(0, MyPageAdapterType.TITLE, profile),
+        AttendanceModel(1, MyPageAdapterType.SCORE, profile)
+    )
+
+    private fun attendanceEmpty(profile: Profile) =
+        myPageHeader(profile) + AttendanceModel(3, MyPageAdapterType.LIST_NONE)
```

Comment:
```suggestion
       val headers  =  myPageHeader(profile)
      return headers  + AttendanceModel(headers.size() + 1, MyPageAdapterType.LIST_NONE)
```

뒤에 인덱스가 myPageHeader의 의존성이 생겨서 위처럼 수정해도 좋을 것 같앙

Reply:
고마워용ㅎㅎ 수정햇어요W<

Reply:
오 지금보니 size + 1 오류인데 그냥 size로 수정해줬네 고마웡